esdb/configuration.go

```diff
@@ -70,6 +70,10 @@ type Configuration struct {
 
 	// The amount of time (in milliseconds) the sender of the keep alive ping waits for an acknowledgement.
 	KeepAliveTimeout time.Duration // Defaults to 10 seconds.
+
+	// The amount of time (in milliseconds) a non-streaming operation should take to complete before resulting in a
+	// DeadlineExceeded. Defaults to 10 seconds.
+	DefaultDeadline *time.Duration
```

Comment:
Is there a need for using a pointer here?
It creates a bit of an inconsistency with `KeepAliveInterval` & `KeepAliveTimeout`

Reply:
To detect if a default timeout was set.